### ISSUE TITLE
Eslint is not marking not defined properties of a component as invalid

### ISSUE THREAD

**Issue by pbrzosko** [vuejs/eslint-plugin-vue#2459]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.24.0
- **eslint-plugin-vue version:** 9.5.1
- **Vue version:** 3.4.16
- **Node version:** 20.11.0
- **Operating System:** OSX Sonoma 14.4.1

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  plugins: ["no-only-tests"],
  env: {
    node: true,
  },
  extends: [
    "plugin:vue/vue3-essential",
    "eslint:recommended",
    "@vue/eslint-config-typescript",
    "@vue/eslint-config-prettier",
  ],
  parserOptions: {
    ecmaVersion: "latest",
  },
  rules: {
    "prettier/prettier": [
      "error",
      {
        endOfLine: "auto",
      },
    ],
    "no-only-tests/no-only-tests": process.env.NODE_ENV === "production" ? "error" : "warn",
    "no-console": process.env.NODE_ENV === "production" ? "error" : "off",
    "no-debugger": process.env.NODE_ENV === "production" ? "error" : "off",
    "@typescript-eslint/consistent-type-imports": "error",
  },
};

```

**What did you do?**	

```vue
<!-- ComponentA -->
<script setup lang="ts">
const props = defineProps({
  message: {
    type: String,
  },
});
</script>
<template>Hello {{ props.message }}</template>

<!-- ComponentB -->
<script setup lang="ts">
import ComponentA from "./ComponentA.vue";
import { ref } from "vue";

const message = ref("Test");
</script>
<template><ComponentA :property-that-doesnt-exist="message" other-property="Value" /></template>

```

**What did you expect to happen?**
I would expect that eslint will mark "property-that-doesnt-exist" and "other-property" as invalid, as they are neither HTML attributes nor properties defined on the component.

**What actually happened?** No error message.

**Repository to reproduce this issue** Code is provided in the issue description.

**Comment by FloEdelmann** [comment]
eslint-plugin-vue doesn't have access to other components' files, so this kind of check is impossible. Also note that `property-that-doesnt-exist` will actually render as an HTML attribute.

**Comment by pbrzosko** [comment]
So, should it be vue-tsc that could handle that?

**Comment by FloEdelmann** [comment]
vue-tsc will also not report this as an error because – as I said – that `property-that-doesnt-exist` is not invalid and will actually render as an HTML attribute.

**Comment by pbrzosko** [comment]
But I guess there is a finite set of HTML attributes plus the ones starting with data-* and aria-*. So in theory you could validate whether provided attribute is a component property or a valid HTML attribute and then have a fail if not?

**Comment by FloEdelmann** [comment]
> But I guess there is a finite set of HTML attributes plus the ones starting with data-* and aria-*.

The `aria-*` are also explicitly defined as far as I know, so e.g. `aria-foo` should then also be reported.

> So in theory you could validate whether provided attribute is a component property or a valid HTML attribute and then have a fail if not?

In vue-tsc, this would potentially be possible. In eslint-plugin-vue, this is infeasible, because we can't distinguish "invalid" HTML properties from valid component props.

**Comment by pbrzosko** [comment]
Ok, please close this one then. I will post one for vue-tsc team.

**Comment by iamandrewluca** [comment]
Hey, @pbrzosko, have you had any success with this problem?

I was thinking maybe using ts tooling for eslint may help with this 🤔

**Comment by pbrzosko** [comment]
Please check uncategirized rules of Vue - I think there is something there about props. I jave recently changed to Vue recommended config and I think there was something about props that don't exist.

**Comment by iamandrewluca** [comment]
I checked all the rules of vue/eslint some time ago and could not find any rule for this.

You gave me a chance again here :D checked again, there are some props related rules, but not for this case :)

Either way thanks for your response.

**Comment by FloEdelmann** [comment]
@iamandrewluca please see [my comment above](https://github.com/vuejs/eslint-plugin-vue/issues/2459#issuecomment-2096065762): This kind of check is impossible with eslint-plugin-vue and typescript-eslint. It is potentially possible with [vue-tsc](https://github.com/vuejs/language-tools/tree/master/packages/tsc) (which is used by the [VS Code Vue extension](https://marketplace.visualstudio.com/items?itemName=Vue.volar), formerly Volar), which already complains about missing required props or wrong types. But passing an additional prop is not an error; it will be rendered as a custom HTML attribute.

**Comment by iamandrewluca** [comment]
Hey, @FloEdelmann, yes, I read entire thread yesterday, I was hoping that something changed in a while :)

> But passing an additional prop is not an error; it will be rendered as a custom HTML attribute.

If such a thing existed, it would be like an explicit mode, where you need to type all the possible props.

Thanks.
